prow/scripts/cluster-integration/helpers/create-dns-record.sh

```diff
@@ -25,23 +25,33 @@ if [ "${discoverUnsetVar}" = true ] ; then
     exit 1
 fi
 
-trap cleanup EXIT
+retries=3
+retryTimeInSec="5"
+function createDNSWithRetries() {
+    set +e
 
-cleanup() {
-    if [ "${CLEANUP_DNS_TRANSACTION}" == true ]; then
-        gcloud dns record-sets transaction abort --zone="${CLOUDSDK_DNS_ZONE_NAME}" --verbosity none
-    fi
-}
+    for ((i=1; i<=retries; i++)); do
+        gcloud dns --project="${CLOUDSDK_CORE_PROJECT}" record-sets transaction start --zone="${CLOUDSDK_DNS_ZONE_NAME}" && \
+        gcloud dns --project="${CLOUDSDK_CORE_PROJECT}" record-sets transaction add "${IP_ADDRESS}" --name="${DNS_FULL_NAME}" --ttl=300 --type=A --zone="${CLOUDSDK_DNS_ZONE_NAME}" && \
+        gcloud dns --project="${CLOUDSDK_CORE_PROJECT}" record-sets transaction execute --zone="${CLOUDSDK_DNS_ZONE_NAME}"
 
-CLEANUP_DNS_TRANSACTION=true
+        [[ $? -eq 0 ]] && break
 
-gcloud dns --project="${CLOUDSDK_CORE_PROJECT}" record-sets transaction start --zone="${CLOUDSDK_DNS_ZONE_NAME}"
+        gcloud dns record-sets transaction abort --zone="${CLOUDSDK_DNS_ZONE_NAME}" --verbosity none
+
+        if [[ "${i}" -lt "${retries}" ]]; then
+            echo "Unable to create DNS record, let's wait ${retryTimeInSec} seconds and retry [${i}/${retries}]."
+        else
+            echo "Unable to create DNS record after ${retries} retries, giving up."
```

Comment:
The script should exit with code 1 if it gives up. Waiting 10 minutes for something that is not going to happen seems pointless.

Reply:
good catch